arch/avr/src/avr/up_dumpstate.c

```diff
@@ -51,7 +52,11 @@ static void up_stackdump(uint16_t sp, uint16_t stack_top)
 {
   uint16_t stack;
 
-  for (stack = sp & ~3; stack < (stack_top & ~0x1f); stack += 12)
+  /* Flush any buffered SYSLOG data to avoid overwrite */
+
+  syslog_flush();
+
+  for (stack = sp & ~3; stack < (stack_top & ~0x3); stack += 12)
```

Comment:
```suggestion
  for (stack = sp & ~0x3; stack < (stack_top & ~0x3); stack += 12)
```
Just to keep same style as in other files